src/redux/data.ts

```diff
@@ -1580,7 +1542,18 @@ export const checkPendingTransactionsOnInitialize = (
     currentAccountAddress,
     'latest'
   );
-  await dispatch(dataWatchPendingTransactions(provider, currentNonce));
+  const notPendingTxs = await dispatch(
+    dataWatchPendingTransactions(provider, currentNonce)
+  );
+  if (!notPendingTxs) {
+    dispatch(
+      watchPendingTransactions(
+        currentAccountAddress,
+        TXN_WATCHER_MAX_TRIES,
+        null
```

Comment:
when there's a pending tx on initialize we're currently taking the mainnet provider, which doesn't makes sense for l2 txs, so now we'll be choosing provider depending on the tx network